Detectors/FIT/FDD/reconstruction/src/Reconstructor.cxx

```diff
@@ -22,25 +22,28 @@
 using namespace o2::fdd;
 
 //_____________________________________________________________________
-o2::fdd::RecPoint Reconstructor::process(o2::fdd::Digit const& digitBC,
-                                         gsl::span<const o2::fdd::ChannelData> inChData,
-                                         gsl::span<o2::fdd::ChannelDataFloat> outChData)
+void Reconstructor::process(o2::fdd::Digit const& digitBC, gsl::span<const o2::fdd::ChannelData> inChData,
+                            std::vector<o2::fdd::RecPoint>& RecPoints, std::vector<o2::fdd::ChannelDataFloat>& outChData)
+// gsl::span<o2::fdd::ChannelDataFloat> outChData)
 {
   // Compute charge weighted average time
   Double_t timeFDA = 0, timeFDC = 0;
   Double_t weightFDA = 0.0, weightFDC = 0.0;
   Int_t nInTimeA = 0, nInTimeC = 0;
-
+  int firstEntry = outChData.size();
+  int nStored = 0;
   int nch = inChData.size();
   for (int ich = 0; ich < nch; ich++) {
-    outChData[ich] = o2::fdd::ChannelDataFloat{inChData[ich].mPMNumber,
-                                               (inChData[ich].mTime) * timePerTDC,
-                                               (double)inChData[ich].mChargeADC,
-                                               inChData[ich].mFEEBits};
-
-    Float_t adc = outChData[ich].mChargeADC;
-    Float_t time = outChData[ich].mTime;
     bool inTime = inChData[ich].getFlag(ChannelData::EEventDataBit::kIsEventInTVDC);
+    bool inAdcGate = inChData[ich].getFlag(ChannelData::EEventDataBit::kIsCFDinADCgate);
+    if (inAdcGate) {
```

Comment:
This condition is OK, but it's quite relaxed. So the question is: do you want to get more precise time sacrificing efficiency, or vice-versa? This requires a lot of testing, starting with simulations, but ultimately with the experimental data. Have you done any tests?